dandi/tests/test_files.py

```diff
@@ -98,6 +113,115 @@ def test_find_dandi_files(tmp_path: Path) -> None:
     ]
 
 
+def test_find_dandi_files_with_bids(tmp_path: Path) -> None:
+    mkpaths(
+        tmp_path,
+        dandiset_metadata_file,
+        "foo.txt",
+        "bar.nwb",
+        "bids1/dataset_description.json",
+        "bids1/file.txt",
+        "bids1/subdir/quux.nwb",
+        "bids1/subdir/glarch.zarr/dataset_description.json",
+        "bids2/dataset_description.json",
+        "bids2/movie.mp4",
+        "bids2/subbids/dataset_description.json",
+        "bids2/subbids/data.json",
+    )
```

Comment:
would be nice if after initial check you extend this with subdataset e.g.
```
"bids1/rawdata/dataset_description.json",
"bids1/rawdata/sub-1/micr/sub-1_micr.ext"
```
and saw that result is identical to prior one but with the new entities added for this. And then add zarr somewhere and see that it adds up as well nicely while the rest of the sorted result stays intact.

Reply:
The tests already have a subdataset (`bids2/subbids/dataset_description.json`) and a Zarr (`bids1/subdir/glarch.zarr/dataset_description.json` — it doubles as a test of handling `dataset_description.json` inside Zarrs).

Reply:
ah, ok, although I would still think "incremental" growth of test case could be of benefit and could possibly cover more codepaths.